openshift.py

```diff
@@ -0,0 +1,750 @@
+
+
+class openshift:
+    headers = None
+    verify = False
+    url = None
+
+    def __init__(self, url, token, logger):
+        self.set_token(token)
+        self.set_url(url)
+        self.logger = logger
+
+    def set_token(self, token):
+        self.headers = {
+            "Authorization": "Bearer " + token,
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def set_url(self, url):
+        self.url = url
+
+    def get_url(self):
+        return self.url
+
+    def cnvt_project_name(self, project_name):
+        suggested_project_name = re.sub("^[^A-Za-z0-9]+", "", project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9]+$", "", suggested_project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9\-]+", "-", suggested_project_name)
+        return suggested_project_name
+
+    def get_request(self, url, debug=False):
+        r = requests.get(url, headers=self.headers, verify=self.verify)
+        if debug == True:
+            self.logger.info("url: " + url)
+            self.logger.info("g: " + str(r.status_code))
+            self.logger.info("g: " + r.text)
+        return r
+
+    def del_request(self, url, payload, debug=False):
+        if payload is None:
+            r = requests.delete(url, headers=self.headers, verify=self.verify)
+        else:
+            r = requests.delete(
+                url, headers=self.headers, data=json.dumps(payload), verify=self.verify
+            )
+        if debug == True:
+            self.logger.info("url: " + url)
+            if payload is not None:
+                self.logger.info("payload:" + json.dumps(payload))
+            self.logger.info("d: " + str(r.status_code))
+            self.logger.info("d: " + r.text)
+        return r
+
+    def put_request(self, url, payload, debug=False):
+        if payload is None:
+            r = requests.put(url, headers=self.headers, verify=self.verify)
+        else:
+            r = requests.put(
+                url, headers=self.headers, data=json.dumps(payload), verify=self.verify
+            )
+        if debug == True:
+            self.logger.info("url: " + url)
+            if payload is not None:
+                self.logger.info("payload:" + json.dumps(payload))
+            self.logger.info("pu: " + str(r.status_code))
+            self.logger.info("pu: " + r.text)
+        return r
+
+    def post_request(self, url, payload, debug=False):
+        r = requests.post(
+            url, headers=self.headers, data=json.dumps(payload), verify=False
+        )
+        if debug == True:
+            self.logger.info("url: " + url)
+            self.logger.info("payload: " + json.dumps(payload))
+            self.logger.info("po: " + str(r.status_code))
+            self.logger.info("po: " + r.text)
+        return r
+
+    def user_rolebinding_exists(self, user, project_name, role):
+        openshift_role = ""
+
+        if role == "admin":
```

Comment:
This mapping of roles to roles is repeated multiple times throughout the file. I would have this be a dictionary at the module level `role_mapping = {'admin': 'admin', 'member': 'edit', 'reader': 'view'}` and then do `if role in role_mapping`.